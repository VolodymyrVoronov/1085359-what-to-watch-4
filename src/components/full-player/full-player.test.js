import React from "react";
import renderer from "react-test-renderer";
import FullPlayer from "./full-player.jsx";

export const CURRENT_MOVIE = {
  id: 0,
  title: `Citizen Kane`,
  genres: [`Documentary`, `Kids & Family`, `Romance`],
  img: `img/aviator.jpg`,
  releaseDate: 2014,

  poster: `img/aviator.jpg`,
  rating: 10,
  ratingReviewsCount: 120,
  description: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Reiciendis natus ipsa ut     possimus laudantium excepturi magni! Soluta doloribus facere quaerat, optio ab ratione quas provident nobis. Accusamus incidunt unde dicta.`,
  story: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Reiciendis natus ipsa ut     possimus laudantium excepturi magni! Soluta doloribus facere quaerat, optio ab ratione quas provident nobis. Accusamus incidunt unde dicta.`,
  director: `Director`,
  actors: [`Actor 1`, `Actor 2`, `Actor 3`, `Actor 4`, `Actor 5`],
  runTime: 1260,
  preview: `http://img1.joyreactor.cc/pics/post/webm/%D0%B3%D0%B8%D1%84%D0%BA%D0%B8-%D1%84%D0%BE%D0%BA%D1%83%D1%81-%D1%86%D0%B2%D0%B5%D1%82%D0%BE%D0%BA-%D0%A0%D0%B5%D0%B9%D0%BC%D0%BE%D0%BD%D0%B4-%D0%A2%D0%B5%D0%BB%D0%BB%D0%B5%D1%80-6063047.webm`,

  reviews: [
    {
      author: `John`,
      score: 5.1,
      text: `It's ok. but it could be better`,
      date: 1582590140667
    },
    {
      author: `Max`,
      score: 4.1,
      text: `It's ok. but it could be better`,
      date: 1582590140667
    },
    {
      author: `Stive`,
      score: 10,
      text: `It's ok. but it could be better`,
      date: 1582590140667
    }
  ]
};
it(`Full Player should render correctly`, () => {
  const tree = renderer
    .create(<FullPlayer
      isPlaying={false}
      progress={10}
      duration={99}
      onPlayButtonClick={() => {}}
      onFullScreenButtonClick={() => {}}
      onExitButtonClick={() => {}}
      film={CURRENT_MOVIE}
    >
      <video />
    </FullPlayer>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
