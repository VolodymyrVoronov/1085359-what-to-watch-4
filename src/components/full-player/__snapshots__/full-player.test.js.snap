// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Full Player should render correctly 1`] = `
<div
  className="player"
>
  <video />
  <button
    className="player__exit"
    onClick={[Function]}
    type="button"
  >
    Exit
  </button>
  <div
    className="player__controls"
  >
    <div
      className="player__controls-row"
    >
      <div
        className="player__time"
      >
        <progress
          className="player__progress"
          max={99}
          value={10}
        />
        <div
          className="player__toggler"
          style={
            Object {
              "left": "10.1010101010101%",
            }
          }
        >
          Toggler
        </div>
      </div>
      <div
        className="player__time-value"
      >
        00:01:29
      </div>
    </div>
    <div
      className="player__controls-row"
    >
      <button
        className="player__play"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="19"
          id="play-s"
          viewBox="0 0 19 19"
          width="19"
        >
          <path
            clipRule="evenodd"
            d="M0 0L19 9.5L0 19V0Z"
            fill="#EEE5B5"
            fillRule="evenodd"
          />
        </svg>
        <span>
          Play
        </span>
      </button>
      <div
        className="player__name"
      >
        The Grand Budapest Hotel
      </div>
      <button
        className="player__full-screen"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="27"
          viewBox="0 0 27 27"
          width="27"
        >
          <use
            xlinkHref="#full-screen"
          />
        </svg>
        <span>
          Full screen
        </span>
      </button>
    </div>
  </div>
</div>
`;
